benchmarks/bench_checkpointing_ssu.py

```
@@ -0,0 +1,1269 @@
+    cupti.activity_disable(cupti.ActivityKind.DRIVER)
+    cupti.activity_disable(cupti.ActivityKind.MEMCPY)
+    cupti.activity_disable(cupti.ActivityKind.MEMSET)
+    cupti.finalize()
+
+    # Build correlation_id → kernel list mapping
+    corr_to_kernels: dict[int, list[tuple[int, int, int]]] = {}
+    for k in kernels:
+        corr_to_kernels.setdefault(k[2], []).append(k)
+
+    # Sort launches by start for binary search
+    sorted_launches = sorted(launches, key=lambda l: l[0])
```

Ruff (0.15.12)

[error] Ambiguous variable name: `l`

(E741)